src/providers/wms/qgswmscapabilities.cpp

```diff
@@ -2157,9 +2157,25 @@ const QgsWmtsTileMatrix *QgsWmtsTileMatrixSet::findOtherResolution( double tres,
   QMap<double, QgsWmtsTileMatrix>::const_iterator it = tileMatrices.constFind( tres );
   if ( it == tileMatrices.constEnd() )
     return nullptr;
-  it += offset;
-  if ( it == tileMatrices.constEnd() )
-    return nullptr;
+  while ( 1 )
+  {
+    if ( offset > 0 )
+    {
+      ++it;
+      --offset;
+    }
+    else if ( offset < 0 )
+    {
+      if ( it == tileMatrices.constBegin() )
+        return nullptr;
+      --it;
+      ++offset;
+    }
+    else
+      break;
 
+    if ( it == tileMatrices.constEnd() )
+      return nullptr;
+  }
```

Comment:
Sorry I should have explained it in the commit message... in Qt4 if the iterator is at the first item, going backwards will set the iterator to be equal to constEnd(). However in Qt5 this is not the case and it would just leave iterator in undefined state. So the main change in the commit is to first check if we are not already at constBegin() before going backwards. Probably this was changed to be in line with behavior of iterators of ordinary c++ containers... not sure.